WordPress/Classes/Utility/PushNotificationsManager.swift

```diff
@@ -101,7 +101,7 @@ final public class PushNotificationsManager: NSObject {
         deviceToken = newToken
 
         // Register against WordPress.com
-        let noteService = NotificationSettingsService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        let noteService = NotificationSettingsService(coreDataStack: ContextManager.sharedInstance())
```

Comment:
Did you consider adding an instance property to store `ContextManager.sharedInstance()`?

I don't think it would make much difference at runtime, given it would still access the shared instance. It might be useful in the future when/if we'll remove all the shared instances accesses in favor of Dependency Injection.

🤷‍♂️

Reply:
I don't want to think too far ahead, even though we know DI is going to happen at some point, but we don't know how it's going to look like, which can affect how the code should be written.

Reply:
Fair enough. 👌